csrc/quantization/compressed_tensors/int8_quant_kernels.cu

```diff
@@ -31,6 +31,43 @@ static inline __device__ int8_t float_to_int8_rn(float x) {
 #endif
 }
 
+static inline __device__ int32_t float_to_int32_rn(float x) {
+#ifdef USE_ROCM
+  static const auto i32_min =
+      static_cast<float>(std::numeric_limits<int32_t>::min());
+  static const auto i32_max =
+      static_cast<float>(std::numeric_limits<int32_t>::max());
```

Comment:
I think there is a hazard here -- `static_cast<float>(std::numeric_limits<int32_t>::max());` is not representable with a float, so this will need to round one way or the other, and may end up being too large to fit in an int32. In that case the static cast may roll over.

Do we have any unit tests that exercise the saturation explicitly?

Reply:
We don't have any unit tests, and this is indeed a bug.

Reply:
The comment looks good now -- I know I accepted already, but if it's not too much effort, could you add a unit test covering this case before we merge it?